text-templates/text-templates.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"os"
+	"text/template"
+)
+
+// This is for creating dynamic content or
+// showing customized output to the user with the text/template package.
+func main() {
```

Comment:
If you understood this example well, then you'll also this bit of code from [a mocking framework I made](https://github.com/kinbiko/mokku/blob/main/cmd/mokku/mokku.go#L19-L30) -- same technology!